ownCloudAppShared/Client/File Lists/ClientDirectoryPickerViewController.swift

```diff
@@ -183,30 +219,70 @@ open class ClientDirectoryPickerViewController: ClientQueryViewController {
 	}
 
 	override open func tableView(_ tableView: UITableView, shouldHighlightRowAt indexPath: IndexPath) -> Bool {
-		if let item : OCItem = itemAt(indexPath: indexPath), allowNavigationFor(item: item) {
+		if hasFavorites, indexPath.section == 0 {
+			return true
+		} else if let item : OCItem = itemAt(indexPath: indexPath), allowNavigationFor(item: item) {
 			return true
 		}
 
 		return false
 	}
 
 	override open func tableView(_ tableView: UITableView, willSelectRowAt indexPath: IndexPath) -> IndexPath? {
-		if let item : OCItem = itemAt(indexPath: indexPath), allowNavigationFor(item: item) {
+		if hasFavorites, indexPath.section == 0 {
+			return indexPath
+		} else if let item : OCItem = itemAt(indexPath: indexPath), allowNavigationFor(item: item) {
 			return indexPath
 		}
 
 		return nil
 	}
 
 	override open func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
-		guard let item : OCItem = itemAt(indexPath: indexPath), item.type == OCItemType.collection, let core = self.core, let path = item.path, let selectButtonTitle = selectButtonTitle, let choiceHandler = choiceHandler else {
-			return
-		}
+		if hasFavorites, indexPath.section == 0 {
+			guard let core = self.core else {
+				return
+			}
+
+			let favoriteQuery = OCQuery(condition: .require([
```

Comment:
@hosy May be refactor as separate method? I mean creation of `QueryFileListTalbleViewController` with corresponding query etc.

Reply:
moved this code to a separate function